File: java/fib/Fibonacci.java
import java.util.*;

public class Fibonacci
{

  public void fib( int limit )
  {
    List results = new ArrayList();
    int current = 1;
    int prev = 1;
    int temp = 0;
    results.add(""+prev);
    results.add(""+current);
    
    while( (current + prev) < limit )
    {
      temp = current;
      current = current + prev;
      prev = temp;
      results.add(""+current);
    }

    for( Iterator iter = results.iterator(); iter.hasNext(); )
    {
      System.out.print( iter.next() + " " );
    }
    System.out.print("\n");
    
  }

  public static void main( String [] args )
  {
    Fibonacci F = new Fibonacci();

    if( args.length != 1 )
    {
      System.out.println( "Usage: java Fibonacci <limit>" );
    }
    else
    {
      try
      {
        F.fib( Integer.parseInt( args[0] ) );        
      }
      catch( Exception e )
      {
        System.out.println( e.toString() );
        e.printStackTrace();
      }
    }
  }

}

File: C++/fib/Fibonacci.cpp
#include <iostream.h>
#include <stdlib.h>

struct fibEle
{
  int val;
  fibEle *next;
};

void fib(int limit);
fibEle *getElement( int value );


fibEle *getElement( int value )
{
  fibEle *temp = new fibEle();
  temp->val = value;
  temp->next = NULL;
  return temp;
}

void fib(int limit)
{
  fibEle *head;
  fibEle *prev;
  fibEle *curr;

  prev = new fibEle;
  prev->val = 1;
  
  curr = new fibEle;
  curr->val = 1;
  prev->next = curr;
  head = prev;
  
  int temp = 0;

  while( (prev->val + curr->val) <= limit )
  {
    temp = prev->val + curr->val;
    prev = curr;
    curr = getElement( temp );
    prev->next = curr;
  } 

  fibEle *deleter = NULL;
  fibEle *printer = head;

  while( printer != NULL )
  {
    deleter = printer;
    cout << printer->val << " ";
    printer = printer->next;
    delete deleter;
  }

  cout << endl;
}

int
main( int argc, char **argv)
{
  int value = 0;
  int retVal = 0;

  if( argc != 2 )
  {
    cout << "Usage: fib.exe <limit>" << endl;
    retVal = 1;
  }
  else
  {
    value = atoi( argv[1] );
    fib(value);
  }

  return retVal;
}
